helix-lsp/src/snippet.rs

```diff
@@ -176,14 +176,18 @@ pub fn into_transaction<'a>(
     }
 
     if let Some(first) = ntabstops.first() {
+        let cursor_offset = insert.chars().count() as i128 - (end_offset - start_offset);
+        let mut extra_offset = start_offset;
```

Comment:
This should start at 0

Reply:
Ah I see the reason is that you changed `offset` (at the top) to start at 0 instead of `start_offset` I missed that.

I guess it doesn't really matter which one of these two starts at `start_offset` and `0` respectively (so you could swap the initial value of those and it would work the same)

Reply:
Yes, exactly. I have a slightly tendency to keep `offset` starting at 0, since (for me at least) `offset` represent just the `insert` offset and it was a bit weird that it was negative. Anyway your call.